Automattic/jetpack
# Premium Content Block: Move to Jetpack

### Changes proposed in this Pull Request:
This PR moves the Premium Content block code from Editing toolkit into Jetpack. It reformats the block code to match the Jetpack block organization and removes any duplicate functionality.

**Note:** In order for users to seamlessly transition from the editing toolkit version to the Jetpack version of this block, the namespace has to remain as `premium-content/container` and not include `/jetpack` (this is handled in this PR). Doing this means we do not have to maintain the old code in Editing Toolkit forever.

Fixes https://github.com/Automattic/view-design/issues/144

### Jetpack product discussion


* Post: p1HpG7-aFT-p2

### Does this pull request change what data or activity we track or use?


* I don't think so, but I might be missing something.

----
### Testing instructions:






**Please test this on a local Jetpack install using Jurassic.tube, and on WordPress.com.**

**Testing a Jetpack Site:**
* Please use https://github.com/Automattic/a8c-wp-env to test as it will make your life much easier.
* Start up a site with `npm run start`, and **DO NOT** include Editing Toolkit (just use Gutenberg, and Jetpack).
* Check out this branch and activate jurassic.tube with `npm run tube:init`
* Add a "Security Daily" Jetpack plan to your jurassic.tube site.
* Create a new post and insert the Premium Content block. Confirm this works.
* Connect to Stripe using the `ryan.ray` test stripe account in the secret store. Confirm this connection works.
* Add a payment plan through the block, confirm this works.
* Add some content to the subscriber and visitor views.
* Publish the page, and confirm you can see only the subscriber content in the front end.
* Open an incognito window, confirm you can see only the visitor content in the front end.
* Click the Subscribe button in the incognito window, confirm you see the correct payment plan amount and payment modal.
* Purchase a subscription. You can use this instructions: PCYsg-lW4-p2#testing
* Confirm that you now see the subscriber content in the front end.

**Testing a WordPress.com Site:**
* Please repeat the above on a WordPress.com site -- with the following steps first
* Use `npm run sync` and select Jetpack
* Use `npm run sync` and select Editing Toolkit
* Open `wp-calypso/apps/editing-toolkit/editing-toolkit-plugin/full-site-editing-plugin.php` and remove `add_action( 'plugins_loaded', __NAMESPACE__ . '\load_premium_content' );` on line 297 so the block doesn't load from Editing Toolkit.
* Wait for everything to sync and go ahead and test as above on a sandboxed WordPress.com site.

**Some other tests to note:**
* Try adding a Premium content block from Editing Toolkit on a sandboxed WordPress.com site, and then load up this PR and make sure it seamlessly transitions to the Jetpack version in the editor and front end.
* Remove your Jetpack plan on your Jurassic.tube site, and confirm the Premium content no longer shows up.
* Load up a free WordPress.com site sandboxed, and confirm you still see the in-editor upgrade nudges.
----


### Privacy Changes:
* This introduces a new cookie that is set to remember if you are a subscriber to the site and show you subscription based content without the need to log in again.

### Proposed changelog entry for your changes:


* WIP
